File: ruby/fibonacci_sequence/src/application.rb
# rbs_inline: enabled

# @rbs init_num: Integer
# @rbs iter: Integer
# @rbs return: Array[Integer]
def fibonacci(init_num:, iter:)
  current_num = init_num
  next_num    = current_num + 1
  Array.new(iter) do
    result_num            = current_num
    current_num, next_num = next_num, current_num + next_num
    result_num
  end
end

File: python/fibonacci_sequence/src/application.py
def fibonacci(*, init_num: int, iter: int) -> list[int]:
    current_num = init_num
    next_num = current_num + 1
    result: list[int] = []
    for _ in range(iter):
        result_num = current_num
        current_num, next_num = next_num, current_num + next_num
        result.append(result_num)
    return result
